content/en/platform/corda/4.8/enterprise/operations/deployment/generating-a-node-dockerform.md

```diff
@@ -0,0 +1,346 @@
+| dbDataVolume           | Path to database files for Postgres Docker image  |
+
+To make the database files persistent across multiple `docker-compose` runs, you must set the `dbDataVolume` parameter. If this variable is commented out, the database files will be removed after every `docker-compose` run.
+
+#### Run the Dockerform task
+
+To run the Dockerform task, follow the steps below.
+
+{{< note >}}
+The node configuration described here is just an example. `Dockerform` allows you specify any number of nodes and you can define their configurations and names as needed.
+{{< /note >}}
+
+1. Open the `build.gradle` file of your CorDapp project and add a new gradle task, as shown in the example below.
+
+{{< note >}}
+Make sure to use Corda gradle plugin version 5.0.10 or above.
+{{< /note >}}
+
+{{% warning %}}
+The Docker image name must be specified by using the `dockerImage` property.
+{{% /warning %}}
+
+```groovy
+task prepareDockerNodes(type: net.corda.plugins.Dockerform, dependsOn: ['jar']) {
+    // set docker image for each node
+    dockerImage = "corda/corda-zulu-java1.8-4.4"
+
+    nodeDefaults {
+        cordapp project(":contracts-java")
+    }
+    node {
+        name "O=Notary,L=London,C=GB"
+        notary = [validating : false]
+        p2pPort 10002
+        rpcSettings {
+            address("localhost:10003")
+            adminAddress("localhost:10023")
+        }
+        projectCordapp {
+            deploy = false
+        }
+        cordapps.clear()
+        sshdPort 2222
+    }
+    node {
+        name "O=PartyA,L=London,C=GB"
+        p2pPort 10002
+        rpcSettings {
+            address("localhost:10003")
+            adminAddress("localhost:10023")
+        }
+        rpcUsers = [[user: "user1", "password": "test", "permissions": ["ALL"]]]
+        sshdPort 2223
+    }
+    node {
+        name "O=PartyB,L=New York,C=US"
+        p2pPort 10002
+        rpcSettings {
+            address("localhost:10003")
+            adminAddress("localhost:10023")
+        }
+        rpcUsers = [[user: "user1", "password": "test", "permissions": ["ALL"]]]
+        sshdPort 2224
+    }
+
+    // This property needs to be outside the node {...} elements
+    dockerImage = "corda/corda-zulu-java1.8-4.8"
+}
+```
+
+{{< note >}}
+If you do not specify the sshd port number for a node, it will use the default value `2222`.
+{{< /note >}}
+
+2. To create the nodes defined in the `prepareDockerNodes` gradle task added in the first step, run the following command in a command prompt or a terminal window, from the root of the project where the `prepareDockerNodes` task is defined:
+
+* Linux/macOS: `./gradlew prepareDockerNodes`
+* Windows: `gradlew.bat prepareDockerNodes`
+
+This command creates the nodes in the `build/nodes` directory. A node directory is generated for each node defined in the `prepareDockerNodes` task. The task also creates a `docker-compose.yml` file in the `build/nodes` directory.
+
+{{< note >}}
```

Comment:
This note is massive. Is it really necessary for it to be a note? Just give it a H2 and take away the note.